frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx

```diff
@@ -333,51 +362,67 @@ export function ObjectDetailView({
-              actionItems={actionItems}
-              canZoom={Boolean(
-                canZoom && (canZoomNextRow || canZoomPreviousRow),
-              )}
+    <>
+      <ObjectDetailContainer wide={hasRelationships} className={className}>
+        {maybeLoading ? (
+          <ErrorWrapper>
+            <LoadingSpinner />
+          </ErrorWrapper>
+        ) : hasNotFoundError ? (
+          <ErrorWrapper>
+            <NotFound message={t`We couldn't find that record`} />
+          </ErrorWrapper>
+        ) : (
+          <ObjectDetailWrapperDiv
+            className="ObjectDetail"
+            data-testid="object-detail"
+          >
+            {showHeader && (
+              <ObjectDetailHeader
+                actionItems={actionItems}
+                canZoom={Boolean(
+                  canZoom && (canZoomNextRow || canZoomPreviousRow),
+                )}
+                objectName={objectName}
+                objectId={displayId}
+                canZoomPreviousRow={!!canZoomPreviousRow}
+                canZoomNextRow={canZoomNextRow}
+                showControls={showControls}
+                viewPreviousObjectDetail={viewPreviousObjectDetail}
+                viewNextObjectDetail={viewNextObjectDetail}
+                closeObjectDetail={closeObjectDetail}
+              />
+            )}
+            <ObjectDetailBody
+              data={data}
               objectName={objectName}
-              objectId={displayId}
-              canZoomPreviousRow={!!canZoomPreviousRow}
-              canZoomNextRow={canZoomNextRow}
-              showControls={showControls}
-              viewPreviousObjectDetail={viewPreviousObjectDetail}
-              viewNextObjectDetail={viewNextObjectDetail}
-              closeObjectDetail={closeObjectDetail}
+              zoomedRow={zoomedRow ?? []}
+              settings={settings}
+              hasRelationships={hasRelationships}
+              onVisualizationClick={onVisualizationClick}
+              visualizationIsClickable={visualizationIsClickable}
+              tableForeignKeys={tableForeignKeys}
+              tableForeignKeyReferences={tableForeignKeyReferences}
+              followForeignKey={onFollowForeignKey}
             />
-          )}
-          <ObjectDetailBody
-            data={data}
-            objectName={objectName}
-            zoomedRow={zoomedRow ?? []}
-            settings={settings}
-            hasRelationships={hasRelationships}
-            onVisualizationClick={onVisualizationClick}
-            visualizationIsClickable={visualizationIsClickable}
-            tableForeignKeys={tableForeignKeys}
-            tableForeignKeyReferences={tableForeignKeyReferences}
-            followForeignKey={onFollowForeignKey}
-          />
-        </ObjectDetailWrapperDiv>
-      )}
-    </ObjectDetailContainer>
+          </ObjectDetailWrapperDiv>
+        )}
+      </ObjectDetailContainer>
+
+      <Modal
+        isOpen={typeof executeActionId === "number"}
```

Comment:
it's either a number or undefined, seems `!!executeActionId` would be enough

Reply:
Your suggestion does not handle `0` correctly.

Reply:
you're right, but it is a real use case?

Reply:
I don't want a future reader to look at this code and wonder why 0 is handled differently than all the other numbers - it looks like a bug.